### PR TITLE
v0.11.1 seems ready

### PR DESCRIPTION
**Pull request by ProofOfKeags**

### CODE REVIEW

File: configurator/src/main.rs [hunk 1]
```diff
@@ -216,6 +257,35 @@ fn main() -> Result<(), anyhow::Error> {
...
+        .parsed_extension(); // oid for subject alternative names
+    match ext {
+        x509_parser::extensions::ParsedExtension::SubjectAlternativeName(names) => {
+            if !(&names.general_names).into_iter().any(|a| match *a {
+                x509_parser::extensions::GeneralName::DNSName(host) => host == tor_address,
+                _ => false,
+            }) {
+                println!("Replacing Certificates");
+                // Delete the tls.key
+                std::fs::remove_file(Path::new("/root/.lnd/tls.key"))?;
```
[the remark this answers is not in the document]
Comment (ProofOfKeags) [reply configurator/src/main.rs:279]: no. Tried that. LND chokes.

File: configurator/src/main.rs [hunk 2]
```diff
@@ -128,18 +147,32 @@ pub struct CipherSeedMnemonic {
 
 fn main() -> Result<(), anyhow::Error> {
     let config: Config = serde_yaml::from_reader(File::open("/root/.lnd/start9/config.yaml")?)?;
+    let alias = match config.alias {
+        None => {
+            let alias_path = Path::new("/root/.lnd/start9/default_alias.txt");
```
[the remark this answers is not in the document]
Comment (ProofOfKeags) [reply configurator/src/main.rs:152]: k

File: Dockerfile [hunk 3]
```diff
@@ -7,7 +7,7 @@ ADD . /root
 
 WORKDIR /root/lnd
 
-RUN make install tags="signrpc walletrpc chainrpc invoicesrpc"
+RUN make -j24 install tags="signrpc walletrpc chainrpc invoicesrpc"
```
[the remark this answers is not in the document]
Comment (ProofOfKeags) [reply Dockerfile:10]: :[

[the remark this answers is not in the document]
Comment (ProofOfKeags) [reply Dockerfile:10]: I'll use nproc

File: configurator/src/.backupignore.template [hunk 4]
```diff
@@ -0,0 +1 @@
+data/graph/mainnet/*
```
Comment (dr-bonez) [inline configurator/src/.backupignore.template:1]: If it's not going into a format string, it shouldn't be .template

**Review by dr-bonez — COMMENTED** [review]
I don't see your restore code?

File: configurator/src/.backupignore.template (hunk 4 above)
Comment (ProofOfKeags) [reply configurator/src/.backupignore.template:1]: ok

File: configurator/src/main.rs [hunk 5]
```diff
@@ -33,16 +41,53 @@ fn parse_quick_connect_url(url: Uri) -> Result<(String, String, String, u16), an
     Ok((user.to_owned(), pass.to_owned(), host.to_owned(), port))
 }
 
+struct SkipNulls(Value);
```
Comment (dr-bonez) [inline configurator/src/main.rs:44]: write a test pls

File: configurator/src/main.rs [hunk 6]
```diff
@@ -216,6 +333,41 @@ fn main() -> Result<(), anyhow::Error> {
             watchtower_client_enabled = config.watchtower_client_enabled,
         )?;
     }
+
+    // TLS Certificate migration from 0.11.0 -> 0.11.1 release (to include tor address)
+    let bs = std::fs::read(Path::new("/root/.lnd/tls.cert"))?;
+    let (_, pem) = pem::parse_x509_pem(&bs)?;
+    let cert = pem.parse_x509()?;
+    let subj_alt_name_oid = "2.5.29.17".parse().unwrap();
```
Comment (dr-bonez) [inline configurator/src/main.rs:341]: link documentation that specifies this oid

File: configurator/src/main.rs [hunk 7]
```diff
@@ -225,6 +377,27 @@ fn main() -> Result<(), anyhow::Error> {
             std::thread::sleep(std::time::Duration::from_secs(1));
         }
     }
+
+    let use_channel_backup_data = match restore_info(Path::new("/root/.lnd"))? {
+        None => Ok(None::<serde_json::Value>),
+        Some(_) => {
+            println!("A");
```
Comment (dr-bonez) [inline configurator/src/main.rs:384]: gut this

File: configurator/src/main.rs [hunk 8]
```diff
@@ -225,6 +377,27 @@ fn main() -> Result<(), anyhow::Error> {
...
+            println!("C");
+            reset_restore(Path::new("/root/.lnd"))?;
+            println!("D");
+            let encoded = base64::encode(bs);
+            Ok::<Option<Value>, std::io::Error>(Some(serde_json::json!({
+                "multi_chan_backup": encoded
+            })))
+        }
+    }?;
+    println!("{:?}", use_channel_backup_data);
```
Comment (dr-bonez) [inline configurator/src/main.rs:399]: and this

File: configurator/src/main.rs [hunk 9]
```diff
@@ -236,15 +409,48 @@ fn main() -> Result<(), anyhow::Error> {
...
-            .arg(format!(
-                "{}",
-                serde_json::json!({
+            .arg({
+                let s = serde_json::to_string(&SkipNulls(serde_json::json!({
                     "wallet_password": base64::encode(&password_bytes),
-                })
-            ))
+                    "recovery_window": config.advanced.recovery_window,})))?;
+                println!("{}", s);
```
Comment (dr-bonez) [inline configurator/src/main.rs:416]: pls

File: configurator/src/main.rs [hunk 10]
```diff
@@ -236,15 +409,48 @@ fn main() -> Result<(), anyhow::Error> {
...
+                        .arg("POST")
+                        .arg("--cacert")
+                        .arg("/root/.lnd/tls.cert")
+                        .arg("--header")
+                        .arg(format!("Grpc-Metadata-macaroon: {}", mac_encoded))
+                        .arg("https://localhost:8080/v1/channels/backup/restore")
+                        .arg("-d")
+                        .arg({
+                            let s = serde_json::to_string(&backups)?;
+                            println!("{}", s);
```
Comment (dr-bonez) [inline configurator/src/main.rs:445]: pls

File: manifest.yaml [hunk 11]
```diff
@@ -1,6 +1,7 @@
+---
 compat: v0
 id: lnd
-version: 0.11.0
+version: 0.11.1
 title: Lightning Network Daemon
 description:
   short: "A complete implementation of a Lightning Network node by Lightning Labs"
```
Comment (dr-bonez) [inline manifest.yaml:7]: add alerts

**Review by dr-bonez — CHANGES_REQUESTED** [review]

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**